source/core/String.cpp

```diff
@@ -96,34 +104,40 @@ void String::AppendEscapeEncoded(const Utf8Char* source, IntIndex len)
     for (IntIndex i = len -1; i >= 0; i--) {
         Utf8Char c = *(begin + i);
         switch (c) {
-        case '\n':
-            *ptr-- = 'n';
-            *ptr-- = '\\';
+        case '\b':  // x08
+            *ptr-- = 'b'; *ptr-- = '\\';
+            break;
+        case '\t':  // x09
+            *ptr-- = 't'; *ptr-- = '\\';
             break;
-        case '\r':
-            *ptr-- ='r';
-            *ptr-- = '\\';
+        case '\n':  // x0A
+            *ptr-- = 'n'; *ptr-- = '\\';
+            break;
+        case '\f':  // x0C
+            *ptr-- = 'f'; *ptr-- = '\\';
+            break;
+        case '\r':  // x0D
+            *ptr-- = 'r'; *ptr-- = '\\';
             break;
-        case '\t':
-            *ptr-- = 't';
-            *ptr-- = '\\';
-           break;
-        case '\f' :
-            *ptr-- = 'f';
-            *ptr-- = '\\';
-           break;
-        case '\b':
-           *ptr-- = 'b';
-           *ptr-- = '\\';
-           break;
         case '\\':
-            *ptr-- = '\\';
-            *ptr-- = '\\';
-           break;
+            *ptr-- = '\\'; *ptr-- = '\\';
+            break;
         case '"':
-            *ptr--  = '\"';
-            *ptr-- = '\\';
-        break;
+            *ptr--  = '"'; *ptr-- = '\\';
+            break;
+        case '\x00': case '\x01': case '\x02': case '\x03': case '\x04': case '\x05': case '\x06': case '\x07':
+            *ptr-- = c + 48; *ptr-- = '0'; *ptr-- = '0'; *ptr-- = '0'; *ptr-- = 'u'; *ptr-- = '\\';
```

Comment:
@cglzaguilar replaced them with the ASCII literal representation with I think does convey the behavior clearer